File: android/decksterlib/src/main/java/no/forse/decksterlib/communication/ConnectMessage.kt
package no.forse.decksterlib.communication

import java.util.UUID

abstract class ConnectMessage {
}

data class HelloSuccessMessage(
    // val playerData: PlayerData,
    var connectionId: String = "",
): ConnectMessage()

data class ConnectFailureMessage(var errorMessage: String = ""): ConnectMessage()

File: src/Deckster.Client/Communication/Handshake/ConnectMessage.cs
using Deckster.Core.Games.Common;
using Deckster.Core.Protocol;

namespace Deckster.Client.Communication.Handshake;

public abstract class ConnectMessage : DecksterMessage;

public class HelloSuccessMessage : ConnectMessage
{
    public PlayerData Player { get; set; }
    public Guid ConnectionId { get; set; }
}

public class ConnectSuccessMessage : ConnectMessage;

public class ConnectFailureMessage : ConnectMessage
{
    public string ErrorMessage { get; init; }
}